scipy/stats/_distn_infrastructure.py

```diff
@@ -2212,21 +2239,36 @@ def fit(self, data, *args, **kwds):
               function to be optimized) and ``disp=0`` to suppress
               output as keyword arguments.
 
+            - method : The method to use. The default is "MLE" (Maximum
+              Likelihood Estimate); "GMM" (Generalized Method of Moments)
+              is also available.
+
+            - order : The number of moments to fit with ``method="GMM"``.
+              This integer must be greater than or equal to the number of
+              non-fixed distribution parameters (including location and scale);
+              the default is the number of non-fixed parameters. This keyword
+              is ignored when ``method="MLE"``.
+
         Returns
         -------
-        mle_tuple : tuple of floats
-            MLEs for any shape parameters (if applicable), followed by those
+        parameter_tuple : tuple of floats
+            Estimates for any shape parameters (if applicable), followed by those
             for location and scale. For most random variables, shape statistics
             will be returned, but there are exceptions (e.g. ``norm``).
 
         Notes
         -----
-        This fit is computed by maximizing a log-likelihood function, with
-        penalty applied for samples outside of range of the distribution. The
-        returned answer is not guaranteed to be the globally optimal MLE, it
+        With ``method="MLE"`` (default),
+        the fit is computed by maximizing a log-likelihood function, with
+        penalty applied for samples beyond the range of the distribution.
+
+        With ``method="GMM"``, the fit is computed by minimizing a weighted
+        norm of the errors between data moments and distribution moments.
+
+        The returned answer is not guaranteed to be globally optimal; it
         may only be locally optimal, or the optimization may fail altogether.
```

Comment:
If the optimization fails will an `MME` specific error be thrown? if so, is it worth noting in docstring?

Reply:
It doesn't look like it. I worked `GMM` into the framework that is used for `MLE` (for better or for worse), and previously it called the optimizer like:
```
        vals = optimizer(func, x0, args=(ravel(data),), disp=0)
        if restore is not None:
            vals = restore(args, vals)
        vals = tuple(vals)
        return vals
```
There is no exception handling here, so exceptions raised would not be specific to `fit` or `MLE`.  There are plenty of things in the existing framework, like this, that might be improved. For instance, it should probably use `minimize` rather than the `fmin` interface to perform optimization, and fixed variables should not be included as decision variables in the objective function. I would need to discuss with @WarrenWeckesser whether we want to overhaul things like this. We're on a timeline for the grant, so we'd want to save these sorts of things for after we're done with the required parts.

Reply:
> It doesn't look like it. I worked `GMM` into the framework that is used for `MLE` (for better or for worse), and previously it called the optimizer like:
> 
> ```
>         vals = optimizer(func, x0, args=(ravel(data),), disp=0)
>         if restore is not None:
>             vals = restore(args, vals)
>         vals = tuple(vals)
>         return vals
> ```
> 
> There is no exception handling here, so exceptions raised would not be specific to `fit` or `MLE`. There are plenty of things in the existing framework, like this, that might be improved. For instance, it should probably use `minimize` rather than the `fmin` interface to perform optimization, and fixed variables should not be included as decision variables in the objective function. I would need to discuss with @WarrenWeckesser whether we want to overhaul things like this. We're on a timeline for the grant, so we'd want to save these sorts of things for after we're done with the required parts.

Ok, I'm not advocating that we increase the scope. We know it is something that might be improved, can always get to it later :).

Reply:
>I would need to discuss with @WarrenWeckesser whether we want to overhaul things like this. We're on a timeline for the grant, so we'd want to save these sorts of things for after we're done with the required parts.

yeah that makes sense for the two of you to decide, it would be nice to swap out the optimizer but that doesn't need to change (it shouldn't) in this PR. 

Also not clear what switching out the optimizer (`fmin` -> `minimize`) would do to backwards compatibility, so another reason to keep as is for this PR.

Reply:
As mentioned above, if a non-finite value is encountered during optimization, I'm raising:
```
ValueError: Method of moments encountered a non-finite distribution moment and cannot continue. Consider trying method='MLE'.
```
I have never seen the optimization raise an exception - it keeps going with non-finite values! So I'm not sure what I would catch and re-raise with a specific message.

Reply:
ok.